doc/Language/compilation.pod6

```diff
@@ -0,0 +1,295 @@
+=item Unicode module names
+=item Modules published under the same names by different authors
+=item Having multiple versions of a module installed
+
+The 26 Latin characters is too restrictive for virtually all real modern languages, including English, which
+has diacritics for many loan words.
+
+With a 1:1 relation between module names and file system paths, you enter a world of pain
+once you try to support Unicode on multiple platforms and file systems.
+
+Then there's sharing module names between multiple authors. This one may or may not work out well in practice.
+I can imagine using it for example for publishing a module with some fix until the original author includes
+the fix in the "official" version.
+
+Finally there's multiple versions. Usually people who need certain versions of modules reach for local::lib or
+containers or some home grown workarounds. They all have their own disadvantages. None of them would be necessary
+if applications could just say, hey I need good old, trusty version 2.9 or maybe a bug fix release of that branch.
+
+If you had any hopes of continuing using the simple name mapping solution, you probably gave up at the
+versioning requirement. Because, how would you find version 3.2 of a module when looking for a 2.9 or higher?
+
+Popular ideas included collecting information about installed modules in JSON files but when those turned out to be
+toe-nail growing slow, text files were replace by putting the meta data into SQLite databases.
+However, these ideas can be easily shot down by introducing another requirement: distribution packages.
+
+Packages for Linux distributions are mostly just archives containing some files plus some meta data.
+Ideally the process of installing such a package means just unpacking the files and updating the central package database.
+Uninstalling means deleting the files installed this way and again updating the package database.
+Changing existing files on install and uninstall makes packagers' lives much harder, so we really want to avoid that.
+Also the names of the installed files may not depend on what was previously installed.
+We must know at the time of packaging what the names are going to be.
+
+=head2 Long names
+
+=begin code
+Foo::Bar:auth<cpan:nine>:ver<0.3>:api<1>
+=end code
+
+Step 0 in getting us back out of this mess is to define a long name.
+A full module name in C<Raku> consists of the short-name, auth, version and API
+
+At the same time, the thing you install is usually not a single module but a distribution which probably contains one or more modules.
+Distribution names work just the same way as module names.
+Indeed, distributions often will just be called after their main module.
+An important property of distributions is that they are immutable.
+C<V< Foo:auth<nine>:ver<0.3>:api<1> >> will always be the name for exactly the same code.
+
+=head2 $*REPO
+In C<Perl> and C<Python> you deal with include paths, pointing to file system directories.
```

Comment:
`$*REPO` is not documented: https://github.com/perl6/doc/issues/502 Maybe it would be a better idea to move this elsewhere to solve that issue, and then close it.